### PR TITLE
Update Trapdoor.php

### PR DESCRIPTION
**Pull request by DavidJBrockway**
Trap Doors now will move the correct way when touched it was a small bug and now it is fixed.

### CODE REVIEW

File: src/pocketmine/block/Trapdoor.php [hunk 1]
```diff
@@ -120,32 +124,40 @@ protected function recalculateBoundingBox(){
...
-			$faces = [
-				2 => 0,
-				3 => 1,
-				4 => 2,
-				5 => 3,
-			];
-			$this->meta = $faces[$face] & 0x03;
+			switch($face){
+				case 2:
+					$this->meta |= 0b00000011;
```
Comment (thebigsmileXD) [inline src/pocketmine/block/Trapdoor.php]: wtf!
look how ClearSky fixed it.

**Comment by DavidJBrockway** [comment]
i fixed it like that

**Comment by DavidJBrockway** [comment]
oit works realy well and tested it

**Comment by DavidJBrockway** [comment]
see i use less code and did some big changes to get it fixed and it works better

**Comment by thebigsmileXD** [comment]
But its a messy code. you do 0b00000000000000000000000000000000000000000000000 (bad usage) instead of (API correct way) 0x00

**Comment by DavidJBrockway** [comment]
I am working on adding red stone soon my first thing to fix is fish and then add different types of doors

**Comment by thebigsmileXD** [comment]
Do you already have ideas about fishing? We are currently stuck on the problem that we don't know how to make the client draw the line, in Bukkit/spigot thats with uuidleast and uuid most.

**Comment by DavidJBrockway** [comment]
Yes I do I am working on it

**Comment by DavidJBrockway** [comment]
Yes I do for mcpe I am working on it also I am working on fixing door bugs and fish item are not the right if you want help plz email me ZenaGamingsky@gmail.com

**Comment by DavidJBrockway** [comment]
Plz email me and I will send you the code I made it sometimes work and it sometime don't it is a glitch I am fixing

**Comment by thebigsmileXD** [comment]
Wait, you sometimes see a line?
Thats cool

**Comment by DavidJBrockway** [comment]
I don't see it to fix it plz email me and. Share my code

**Comment by DavidJBrockway** [comment]
i made a pull with the code but we must add some more code in player of entrie i think

**Pull request closed without merging**